### DISCUSSION TITLE
How to get a contract instance type?

### DISCUSSION THREAD

**Discussion by Ncookiez** [wevm/viem#544] [Question] [answered]
Hi! Currently migrating over from Ethers, where a `new Contract()` would return an object of type `Contract`.

I can't seem to get proper typing for the equivalent `getContract()` function.

### Old Ethers Setup:

```ts
let contract: Contract | undefined
contract = new Contract( ... )
```

### Viem Setup:

```ts
let contract: any
contract = getContract({ ... })
```

---

What should I use there instead of `any`? Does Viem export any contract typing? I see `GetContractReturnType` is exported, but doesn't seem very useful without the rest of the client and ABI typing used internally.

**Comment by rhubarbgarden** [comment, marked as answer]
I realise this is quite an old question, but the way I solved for this was to do `GetContractReturnType<typeof ContractAbi, PublicClient, WalletClient>`. This requires that you create a type definition file for your `ContractAbi`.

**Reply by killroy192** [reply]
GetContractReturnType<typeof abi, PublicClient | WalletClient> worked for v2.17.3
